File: part_1/random_gen.py
import time
import random
import re


def save_data(data: str) -> None:
    with open("python_gen.txt", "w", encoding="utf-8") as file:
        file.write(data)


def random_gen() -> str:
    data = ""
    random.seed(time.time())
    for i in range(128):
        data += str(random.randint(0, 1))
    return data


def main() -> None:
    try:
        generated = random_gen()
        save_data(generated)
    except Exception as ex:
        print("Ошибка: ", ex)


if __name__ == "__main__":
    main()

File: part_1/random_gen.cpp
#include <fstream>
#include <cstdlib>
#include <ctime>
#include <vector>

std::vector<int> random_gen()
{
    std::vector<int> data;
    srand(time(NULL));
    for (int i = 0; i < 128; i++)
    {
        int random = rand() % 2;
        data.push_back(random);
    }
    return data;
}

void save_data(std::vector<int> data)
{
    std::ofstream os;
    os.open("c_gen.txt");
    if (os.is_open())
    {
        for (size_t i = 0; i < data.size(); ++i)
        {
            os << data[i];
        }
    }
    os.close();
}

int main()
{
    std::vector<int> generated = random_gen();
    save_data(generated);
}